### PR TITLE
Rewrite linux useradd provider

### PR DESCRIPTION
**Pull request by lamont-granquist**
Signed-off-by: Lamont Granquist <lamont@scriptkiddie.org>

### CODE REVIEW

**Comment by lamont-granquist** [comment]
:ragecode:

**Comment by thommay** [comment]
👍

**Comment by lamont-granquist** [comment]
bad news is that our internal ci is still busted even after this...  sigh...

**Comment by jkeiser** [comment]
@lamont-granquist I have that on my plate. I think it's a leftover process interacting badly with the current one, but I want to put more diagnostics in there to confirm.

**Comment by lamont-granquist** [comment]
heh, no actually after all this work it was just 834ef1b18ac88c1c848ee768a0a770eeb793f7b7 that was busted

File: lib/chef/resource/user/linux_user.rb [hunk 1]
```diff
@@ -32,7 +32,7 @@ def initialize(name, run_context = nil)
             manage_home: true,
             non_unique: true,
           }
-          @manage_home = true
+          @manage_home = false
```
Comment (coderanger) [inline lib/chef/resource/user/linux_user.rb]: Was this a compat break? https://github.com/chef/chef/pull/5122#r71311061

**Pull request merged**